webroot/src/router/index.ts

```diff
@@ -29,6 +29,16 @@ const router = createRouter({
 router.beforeEach(async (to, from, next) => {
     const isAuthGuardedRoute = to.matched.some((route) => route.meta.requiresAuth);
 
+    // If the store does not have the current compact, set it from the route (e.g. page refreshes)
+    if (to.params?.compact) {
+        const { currentCompact } = store.getters['user/state'];
+
+        if (!currentCompact) {
+            store.dispatch('user/setCurrentCompact', to.params.compact);
+        }
```

Comment:
In what scenarios might the store contain a compact and the query parameters to include one, too?

Reply:
On any normal route nav that's likely to be the case. On manual page refreshes (or entering from a link / bookmark) it's likely the store won't have the compact yet.